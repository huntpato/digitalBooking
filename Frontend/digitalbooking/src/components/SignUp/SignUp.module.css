
.signup{
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}

.button {
  background-color: var(--principal-color);
  border: none;
  border-radius: 5px;
  color: var(--background-white);
  height: 30px;
  margin-top: 20px;
  transition: all 0.3s;
  width: 225px;
}

.button:hover {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
}

.button:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.98);
}

.create {
  display: flex;
  flex-direction: row;
}

.create h5 {
  color: var(--principal-color);
  font-size: 12px;
  padding: 6px 0 0 4px;
  text-decoration: none;
}

.fullname {
  display: flex;
  flex-direction: row;
}

.signup {
  align-items: center;
  background-color: var(--background-color2);
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: 93vh;
  justify-content: center;
  padding-top: 80px;
}

.signup h2 {
  color: var(--principal-color);
  padding-bottom: 10px;
}

.item {
  padding-top: 0px;
}

/*.item input {
  border: var(--background-white);
  -webkit-box-shadow: 0px 3px 5px 0px rgba(243, 241, 237, 0.5);
  box-shadow: 0px 3px 5px 0px rgba(97, 97, 96, 0.5);
}*/

.item input {
  border-radius: 5px;
  border: none;
  border: var(--background-white);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.05);
  height: 25px;
  margin-left: 5px;
  padding: 15px;
  width: 385px;
}

.error {
  color: red;
  padding: 4px 0 0 8px;
}

.error h6 {
  color: red;
  padding: 4px 0 0 8px;
}


.signup section input {
  border: none;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.05);
  width: 190px;
}

i {
  color: gray;
  cursor: pointer;
  margin-right: 10px;
}

.signup p {
  font-size: 13px;
  padding: 5px 0px 5px 8px;
}

.errorm{
  color:red;
  font-size: 12px;
  padding-top: 20px;
}



@media (max-width: 480px) {


  .fullname {
    flex-direction: column;
  }

  .signup input {
    width: 300px;
  }

  .signup section input {
    width: 300px;
  }

  .button {
    width: 300px;
  }
}

@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

