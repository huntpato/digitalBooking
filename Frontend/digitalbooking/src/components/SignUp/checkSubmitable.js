export default function checkSubmitable(values) {
  if(values.password.length >= 6){
    if (values.nombre.trim() && values.apellido.trim() && values.email.trim() && values.password.trim() && values.repassword.trim() && values.repassword === values.password && /\S+@\S+\.\S+/.test(values.email)){
        return true;
    }else{
        return false;
    }    
} else{
    return false;
}  
}
  