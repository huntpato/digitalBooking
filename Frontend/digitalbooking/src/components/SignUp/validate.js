export default function validate(values) {
  let errors = {};

  if (!values.nombre.trim()) {
    errors.nombre = "Este campo es requerido";
  }

  if (!values.apellido.trim()) {
    errors.apellido = "Este campo es requerido";
  }

  if (!values.email) {
    errors.email = "Este campo es requerido";
  } else if (!/\S+@\S+\.\S+/.test(values.email)) {
    errors.email = "Ingrese un email v치lido";
  }

  if (!values.password) {
    errors.password = "Este campo es requerido";
  } else if (values.password.length < 6) {
    errors.password = "La contrase침a debe tener m치s de 6 caracteres";
  }

  if (!values.repassword) {
    errors.repassword = "Este campo es requerido";
  } else if (values.repassword !== values.password) {
    errors.repassword = "Las contrase침as no coinciden";
  }
  return errors;
}
