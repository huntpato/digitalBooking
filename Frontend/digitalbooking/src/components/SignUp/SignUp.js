import React from "react";
import { useHistory } from "react-router-dom";
import { Link } from "react-router-dom"
import { useState, useEffect } from "react";
import validate from "./validate";
import { LoginAnimation } from "../LoginAnimation/LoginAnimation";
import { useForm } from "../../hooks/useForm";
import checkSubmitable from "./checkSubmitable"
import styles from "./SignUp.module.css";

export const SignUp = (props) => {
  const { signup, item, fullname, input, button, create, error, errorm } = styles;
  const url = 'https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/users/save';
  const [success,  setsuccess] = useState(false);
  const [errors, seterrors] = useState([]);
  const [submitable, setsubmitable] = useState(false);
  const [formvalues, handleInputChange] = useForm({
    nombre: "",
    apellido: "",
    email: "",
    password: "",
    repassword: "",
    userRole: "USER",
  });
  const { nombre, apellido, email, password, repassword } = formvalues;
  const [showPasswordA, setShowPasswordA] = useState(false);
  const [showPasswordB, setShowPasswordB] = useState(false);
  const [message, setmessage]= useState();
  let history = useHistory();

  const togglePasswordA = () => {
    setShowPasswordA(showPasswordA ? false : true);
  };

  const togglePasswordB = () => {
    setShowPasswordB(showPasswordB ? false : true);
  };

  useEffect(() => {
    setsubmitable(checkSubmitable(formvalues))
   
  }, [formvalues])

  const handleSubmit = (e) => {
    e.preventDefault();
    setmessage();
    seterrors(validate(formvalues));
    if (submitable) {
      fetch(url, {
        method: "POST",
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json'
        },
        body: JSON.stringify(formvalues),
      })
      .then(response => {
        if (!response.ok) throw Error(response.status);
        return response;
      })
      .then(response => {
        setmessage("Usuario registrado exitosamente. Redirigiendo al home");
        setsuccess("true");
        setTimeout(() => {
           history.push("/");
        }, 2000);
      })
      .catch(error => { 
        error.message == "409"? setmessage("Email registrado previamente."):
        setmessage("Error al realizar el registro.");
        setsuccess(false) 
      }); 
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div className={signup}>
        <h2>Crear cuenta</h2>
        <section className={fullname}>
          <div className={item}>
            <p>Nombre</p>
            <input
              name="nombre"
              className={input}
              type="text"
              value={nombre}
              onChange={handleInputChange}
            />
            {errors.nombre && <h6 className={error}>{errors.nombre}</h6>}
          </div>
          <div className={item}>
            <p>Apellido</p>
            <input
              name="apellido"
              className={input}
              type="text"
              value={apellido}
              onChange={handleInputChange}
            />
            {errors.apellido && <h6 className={error}>{errors.apellido}</h6>}
          </div>
        </section>
        <div className={item}>
          <p>Correo electrónico</p>
          <input
            name="email"
            className={input}
            type="text"
            value={email}
            onChange={handleInputChange}
          />
          {errors.email && <h6 className={error}>{errors.email}</h6>}
        </div>
        <div className={item}>
          <p>Contraseña</p>
          <input
            name="password"
            className={input}
            type={showPasswordA ? "text" : "password"}
            value={password}
            onChange={handleInputChange}
          />
          <i
            className="fas fa-eye-slash"
            onMouseDown={togglePasswordA}
            onMouseUp={togglePasswordA}
          ></i>
          {errors.password && <h6 className={error}>{errors.password}</h6>}
        </div>
        <div className={item}>
          <p>Confirmar contraseña</p>
          <input
            name="repassword"
            className={input}
            type={showPasswordB ? "text" : "password"}
            value={repassword}
            onChange={handleInputChange}
          />
          <i
            className="fas fa-eye-slash"
            onMouseDown={togglePasswordB}
            onMouseUp={togglePasswordB}
          ></i>
          {errors.repassword && <h6 className={error}>{errors.repassword}</h6>}
        </div>
        { success ? <LoginAnimation message= { message }/>: <h6 className= {errorm}>{message}</h6>}
        <button name="submit" className={button}>Registrate</button>
        <div className={create}>
          <p>¿Ya tienes cuenta?</p>
          <Link to="/login" style={{textDecoration: 'none'}}><h5>Inicia sesión</h5></Link>
        </div>
      </div>
    </form>
  );
};