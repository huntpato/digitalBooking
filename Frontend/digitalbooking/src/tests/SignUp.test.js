import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from '../components/SignUp/SignUp'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <SignUp />', () => {

    const handleSubmit = jest.fn();
    const wrapper= shallow(<SignUp onSubmit= {handleSubmit}/>);
   
    test('Comprobar que <SignUp /> renderiza correctamente.', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div').at(0);
        expect(div.hasClass('signup')).toBe(true);
    })
    
    test('Verificar que no se postea informacion con submit', ()=>{

        wrapper.find('form').simulate('submit', { preventDefault(){}})
    })

    test('Debe renderizar el input nombre', () => {
      
        expect(wrapper.find('input[name="nombre"]').exists()).toBe(true);

    });

    test('Debe renderizar el input apellido', () => {
      
        expect(wrapper.find('input[name="apellido"]').exists()).toBe(true);

    });

    test('Debe renderizar el input email', () => {
      
        expect(wrapper.find('input[name="email"]').exists()).toBe(true);

    });

    
    test('Debe renderizar el input password', () => {
      
        expect(wrapper.find('input[name="password"]').exists()).toBe(true);

    });

    test('Debe renderizar el input repassword', () => {
      
        expect(wrapper.find('input[name="repassword"]').exists()).toBe(true);

    });

    test('Debe renderizar el boton de submit', () => {
      
        expect(wrapper.find('button[name="submit"]').exists()).toBe(true);

    });

    test('Verificar que los inputs se renderizan vacios', () => {
    
        expect(wrapper.find('input[name="nombre"]').prop('value')).toBe('');
        expect(wrapper.find('input[name="apellido"]').prop('value')).toBe('');
        expect(wrapper.find('input[name="email"]').prop('value')).toBe('');
        expect(wrapper.find('input[name="password"]').prop('value')).toBe('');
        expect(wrapper.find('input[name="repassword"]').prop('value')).toBe('');
    });
    
});
    
