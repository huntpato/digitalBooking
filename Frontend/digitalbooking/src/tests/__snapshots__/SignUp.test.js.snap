// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <SignUp /> Comprobar que <SignUp /> renderiza correctamente. 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="signup"
  >
    <h2>
      Crear cuenta
    </h2>
    <section
      className="fullname"
    >
      <div
        className="item"
      >
        <p>
          Nombre
        </p>
        <input
          className="input"
          name="nombre"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="item"
      >
        <p>
          Apellido
        </p>
        <input
          className="input"
          name="apellido"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </section>
    <div
      className="item"
    >
      <p>
        Correo electrónico
      </p>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="item"
    >
      <p>
        Contraseña
      </p>
      <input
        className="input"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <i
        className="fas fa-eye-slash"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      />
    </div>
    <div
      className="item"
    >
      <p>
        Confirmar contraseña
      </p>
      <input
        className="input"
        name="repassword"
        onChange={[Function]}
        type="password"
        value=""
      />
      <i
        className="fas fa-eye-slash"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      />
    </div>
    <h6
      className="errorm"
    />
    <button
      className="button"
      name="submit"
    >
      Registrate
    </button>
    <div
      className="create"
    >
      <p>
        ¿Ya tienes cuenta?
      </p>
      <Link
        style={
          Object {
            "textDecoration": "none",
          }
        }
        to="/login"
      >
        <h5>
          Inicia sesión
        </h5>
      </Link>
    </div>
  </div>
</form>
`;
